internal/action/actions.go

```diff
@@ -490,38 +490,84 @@ func (h *BufPane) SelectToEndOfLine() bool {
 	return true
 }
 
-// ParagraphPrevious moves the cursor to the previous empty line, or beginning of the buffer if there's none
-func (h *BufPane) ParagraphPrevious() bool {
+func (h *BufPane) paragraphPrevious() {
 	var line int
+	// Skip to the first non-empty line
 	for line = h.Cursor.Y; line > 0; line-- {
+		if len(h.Buf.LineBytes(line)) != 0 {
+			break
+		}
+	}
+	// Find the first empty line
+	for ; line > 0; line-- {
 		if len(h.Buf.LineBytes(line)) == 0 && line != h.Cursor.Y {
```

Comment:
`line != h.Cursor.Y` seems not needed anymore?

Reply:
You're right, thanks.